### DISCUSSION TITLE
Use specific vercel CLI version for serverless function on live

### DISCUSSION THREAD

**Discussion by EresDev** [vercel/community#2321] [Help] [answered]
### Summary

Recently, Vercel CLI has started to give us a really hard time. Different version, give different errors and it is a nightmare. 

One quick fix for us is to use a specific CLI version. Is there any way to set this on Vercel for serverless functions?

### Additional information

Have a look at the extremely useless logs I am getting? Almost zero information. 
```

RequestId: 6304ca8a-256d-431a-91cf-8bc80718b875 Error: Runtime exited with error: exit status 1
Runtime.ExitError
```

**Comment by amyegan** [comment, marked as answer]
Locking the CLI version is a possible workaround, but not a good longterm solution since it can lead to other problems. Can you share more about your project and the other error logging you've seen? I would like to help you find a more solid solution.

**Reply by EresDev** [reply]
First tell me how to lock it? because we have things things to launch and once we  launched them, they were not working. We we have been fighting with own stuff for days, and turned out nothing we build had issues, but it was the vercel who silently changed its CLI version in background, giving zero warning and useless error.

**Reply by amyegan** [reply]
@EresDev I understand you are frustrated, and I'm sorry you're running into build errors. You can try adding an environment variable with the name `VERCEL_CLI_VERSION` and a value indicating your desired version number, such as `vercel@29.0.2`.

Please do let me know the CLI version number that solves your problem as well as the framework used and any other information you're comfortable sharing. This information will be very helpful for investigating and fixing the problem.

**Reply by EresDev** [reply]
Thank you. It helped. I spent 3 days on this and the biggest problem on your side is bad error messages. There is nothing in the error messages. Even possibility of CLI version was  my guess. Those error messages tell nothing. I remember they used to be very useful when started developing it.

Your CLI version change usually bundles up when we push some new feature and when it fails with such useless error message we end up looking more into our new feature instead of CLI problem and waste a lot of time. 

Please improve the error messages. Show what is actually wrong. 

Other than this, we have a React 16.13.1 app. Uses node 14.14.0. There are serverless functions with it in typescript and javascript. vercel cli 27.4.0 works fine. But with vercel cli above v29, it is a hell house.

**Comment by EresDev** [comment]
You guys are done really things to error logs. I remember, it used to give very details message. Now, I only get this. 

`
RequestId: 6304ca8a-256d-431a-91cf-8bc80718b875 Error: Runtime exited with error: exit status 1
Runtime.ExitError`

Anyways. some success. In vercel serverless files all .js files work fine. It is .ts file which are not working. and giving above error on api request.

**Comment by EresDev** [comment]
Changing CLI version when someone has just pushed their own new changes is such a bad idea. It should be not be bundled with things people are already trying to fix.

**Comment by xts-bit** [comment]
@EresDev @amyegan I'm facing the same issue but your solution is not working in my case, I'm using node v20 in my local machine and it works fine 

but deploying code on Vercel gives the same error I'm using Node.js with Express.js with TypeScript, 

Any solution, please?

**Comment by teppakornt** [comment]
Same issues, I already updated the latest version of Vercel and got the message with **only 1 sentence**.
It doesn't work for me.

```
RequestId: 07e754b3-9417-425f-a841-8a290b0b97e4 Error: Runtime exited with error: exit status 1
Runtime.ExitError

```
My node version on local and server is 18
Vercel-cli version is 32.3.1
